src-tauri/src/common.rs

```diff
@@ -0,0 +1,182 @@
+        let output_str = String::from_utf8_lossy(&output.stdout);
+        // In case the tracked app is restarted, lsof prints two PIDS and the correct PID
+        // is typically the last end line-separated token.
+        if let Some(pid_str) = output_str.lines().last() {
+            if let Ok(pid) = pid_str.parse::<u32>() {
+                println!("{:?}", pid);
+                return Some(pid);
+            }
+        }
+        None
+    }
+
+    #[cfg(target_os = "windows")]
+    {
+        // Use netstat to list all TCP/UDP connections with PIDs.
+        let output = Command::new("netstat").args(["-ano"]).output().ok()?;
+
+        let stdout = String::from_utf8_lossy(&output.stdout);
+
+        // Look for lines containing “:<port>”
+        for line in stdout.lines() {
+            if line.contains(&format!(":{}", port)) {
+                // The PID is typically the last whitespace-separated token.
+                if let Some(pid_str) = line.split_whitespace().last() {
+                    if let Ok(pid) = pid_str.parse::<u32>() {
+                        return Some(pid);
+                    }
+                }
+            }
+        }
+        None
+    }
+}
+
+/// Retrieves the process start time for a given PID, formatted as a human-readable string.
+///
+/// # Arguments
+///
+/// * `pid` – Process identifier (u32).
+///
+/// # Returns
+///
+/// * `Some(String)` containing the start time in the format `DD/MM/YYYY HH:MM:SS`, or
+/// * `None` if no process with that PID exists.
+pub fn get_process_start_time(pid: u32) -> Option<String> {
+    // Initialize a sysinfo System instance and gather all data.
+    let mut sys = System::new_all();
+    sys.refresh_all();
+
+    let local = Local;
+
+    // Look up the process by PID.
+    if let Some(process) = sys.process(Pid::from_u32(pid)) {
+        // Convert the process start time (seconds since epoch) into a formatted string.
+        let date: String = DateTime::from_timestamp(process.start_time() as i64, 0)?.with_timezone(&local)
+            .format("%d/%m/%Y %H:%M:%S")
+            .to_string();
+        return Some(date);
+    }
+
+    // PID not found in the system
+    None
+}
+
+/// Renames keys in the given database guard by replacing the `old_title` prefix
+/// in keys with the `new_title` prefix. The function iterates over all keys,
+/// collects the keys that start with `old_title.`, and renames them accordingly.
+///
+/// # Arguments
+///
+/// * `guard` - A mutable reference to a writable database guard holding a `HashMap`
+///             where keys are strings and values are of generic type `T`.
+/// * `new_title` - The new title string to replace the old title prefix in keys.
+/// * `old_title` - The old title string prefix to be replaced in keys.
+///
+/// # Type Parameters
+///
+/// * `T` - The type of the values in the hashmap. Must implement `Serialize` and `Debug`.
+pub fn rename_database_keys<T: Serialize + Debug>(
```

Comment:
Daca putem adauga la genul asta de functii si o linie in comentariu de unde este folosita functia sau usecase-ul general. Adica de exemplu: functia este folosita in momentul in care se primesc nu stiu ce date de la aplicatii si trebuie sa redenumim anumite chestii